hack/quickstart/init-master.sh

```diff
@@ -70,31 +70,36 @@ function init_master_node() {
     exit 1
 }
 
-# This script can execute on a remote host by copying itself + kubelet service unit to remote host.
+# This script can execute on a remote host by copying itself + kubelet service unit + auto_approver to remote host.
 # After assets are available on the remote host, the script will execute itself in "local" mode.
 if [ "${REMOTE_HOST}" != "local" ]; then
     # Set up the kubelet.service on remote host
     scp -i ${IDENT} -P ${REMOTE_PORT} ${SSH_OPTS} kubelet.master core@${REMOTE_HOST}:/home/core/kubelet.master
     ssh -i ${IDENT} -p ${REMOTE_PORT} ${SSH_OPTS} core@${REMOTE_HOST} "sudo mv /home/core/kubelet.master /etc/systemd/system/kubelet.service"
 
-    # Copy self to remote host so script can be executed in "local" mode
+    # Copy the auto approver to remote host and execute it for 1h
+    scp -i ${IDENT} -P ${REMOTE_PORT} ${SSH_OPTS} ../auto_approver.sh core@${REMOTE_HOST}:/home/core/auto_approver.sh
+    ssh -i ${IDENT} -p ${REMOTE_PORT} ${SSH_OPTS} core@${REMOTE_HOST} "/home/core/auto_approver.sh 3600" &
```

Comment:
> regardless if it is the first or the 10th node

We may have to special-case at least the first node (e.g. bootkube auto-approves first node). Otherwise nothing starts (no node is available for scheduling, so no pods run).

Another option here is when switching to using static pods for the bootstrap control plane (https://github.com/kubernetes-incubator/bootkube/issues/168) - One other pod could be the initial approver (e.g. the approval mechanism is the ability to prove that it's running on the same node).

> I believe we should not have an auto-approver by default in Bootkube elsewhere than in the hack/ examples.

I agree -- but even in those examples I want to be moving away from on-host configuration (which this script ultimately is). Running it as a backgrounded bash job also makes this more difficult to debug.

> We could however simply have that script as a systemd unit on masters?

Same issue - we should move away from on-host configuration (e.g. no new systemd units unless unavoidable). Maybe we change bootkube to auto approve X nodes, or all nodes while running, or? Past that point I think we either need to provide example approvers which can run as pods in the cluster (maybe just an example "approve all") & documentation (curl?) on how to approve new nodes manually.

> Or maybe bootkube should have an integrated command to manage approvals

I want to keep bootkube as a "run only once" tool, after bootstrap we should only require using pure upstream tooling. In a pinch, maybe curl + good docs (and putting effort into getting necessary features into kubeadm).